src/Microsoft.AspNetCore.OData/Query/Expressions/QueryBinderContext.cs

```diff
@@ -235,7 +235,18 @@ public void RemoveParameter(string name)
                 }
             }
 
-            ParameterExpression parameter = Expression.Parameter(Model.GetClrType(edmTypeReference, AssembliesResolver), rangeVariable.Name);
+            Type clrType = null;
+            if (edmTypeReference != null)
+            {
+                clrType = Model.GetClrType(edmTypeReference, AssembliesResolver);
+            }
+            else
+            {
+                // Edm type reference will be null in a dynamic property scenario
+                clrType = typeof(object);
+            }
+
```

Comment:
```suggestion
            // Edm type reference will be null in a dynamic property scenario hence we use the object type.
            Type clrType = null = edmTypeReference != null ? Model.GetClrType(edmTypeReference, AssembliesResolver) : typeof(object)
```
nit: Not sure if this is better but it compresses the initialization to a much smaller text window for readability, though ternary operator can also be less readable in some cases.